src/Mvc/Mvc.ViewFeatures/src/RazorComponents/ComponentRenderer.cs

```diff
@@ -87,6 +88,11 @@ private async Task<IHtmlContent> StaticComponentAsync(HttpContext context, Type
 
         private async Task<IHtmlContent> PrerenderedServerComponentAsync(HttpContext context, ServerComponentInvocationSequence invocationId, Type type, ParameterView parametersCollection)
         {
+            if (!context.Response.HasStarted)
+            {
+                context.Response.Headers[HeaderNames.CacheControl] = "no-cache, no-store, must-revalidate, max-age=0";
```

Comment:
If you felt like listening to MDN: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control#preventing_caching.

Reply:
I looked at MDN, I was planning to summon @Tratcher here for the exact incantation.

My only goal here is to make sure that we put the most restrictive cache header possible, since we had issues in the past where customers complained when we overrode their cache header for one "less strict", so I want to make sure this doesn't happen.

Reply:
The docs say must-revalidate is redundant with no-cache and max-age=0, we don't use it anywhere else.
```suggestion
                context.Response.Headers[HeaderNames.CacheControl] = "no-cache, no-store, max-age=0";
```

Reply:
I think I remember some issue where people complained about the browser not treating that way, but at this point I'm fine with it, so I'll make the change and get this merged.

Thanks!